File: Core/Entities/Vehicle.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Core.Entities
{
    public class Vehicle
    {
        public int Id { get; set; }

        public int ModelId { get; set; }
        public Model Model { get; set; }

        public bool IsRegistered { get; set; }

        [Required]
        [StringLength(255)]
        public string ContactName { get; set; }

        [StringLength(255)]
        public string ContactEmail { get; set; }

        [Required]
        [StringLength(255)]
        public string ContactPhone { get; set; }

        public DateTime LastUpdate { get; set; }
    }
}

File: ClientApp/src/app/models/Vehicle.ts
import { Model } from "./Model";
import { VehicleFeature } from "./VehicleFeature";

export class Vehicle {
  id: number;

  model: Model;
  modelId: number;

  isRegistered: boolean;

  contactName: string;
  contactEmail: string;
  contactPhone: string;

  lastUpdate: Date;

  vehicleFeature: VehicleFeature[];
}